apps/api/v1/test/lib/bookings/_post.test.ts

```diff
@@ -181,8 +183,38 @@ describe.skipIf(true)("POST /api/bookings", () => {
 
         await handler(req, res);
         console.log({ statusCode: res._getStatusCode(), data: JSON.parse(res._getData()) });
+        createdBooking = JSON.parse(res._getData());
+        expect(prismaMock.booking.create).toHaveBeenCalledTimes(1);
+      });
+
+      test("Reschedule created booking", async () => {
+        const { req, res } = createMocks<CustomNextApiRequest, CustomNextApiResponse>({
+          method: "POST",
+          body: {
+            name: "testReschedule",
+            start: dayjs().add(2, "day").format(),
+            end: dayjs().add(3, "day").format(),
+            eventTypeId: 2,
+            email: "test@example.com",
+            location: "Cal.com Video",
+            timeZone: "America/Montevideo",
+            language: "en",
+            customInputs: [],
+            metadata: {},
+            userId: 4,
+            rescheduleUid: createdBooking.uid,
+          },
+          prisma,
+        });
 
+        prismaMock.eventType.findUniqueOrThrow.mockResolvedValue(buildEventType());
+        prismaMock.booking.findMany.mockResolvedValue([]);
+
+        await handler(req, res);
+        console.log({ statusCode: res._getStatusCode(), data: JSON.parse(res._getData()) });
+        const rescheduledBooking = JSON.parse(res._getData()) as Booking;
         expect(prismaMock.booking.create).toHaveBeenCalledTimes(1);
+        expect(rescheduledBooking.fromReschedule).toEqual(createdBooking.uid);
```

Comment:
Using prismaMock we can access createdBooking in memory after re-scheduling, right? If so, I would add a check if `status` of the createdBooking has been set to `cancelled` after rescheduling.